libmscore/textbase.cpp

```diff
@@ -2476,12 +2476,48 @@ int TextBase::getPropertyFlagsIdx(Pid id) const
       return -1;
       }
 
+//---------------------------------------------------------
+//   offsetTid
+//---------------------------------------------------------
+
+Sid offsetTid(Tid tid, bool placeAbove)
+      {
+      switch (tid) {
+            case Tid::DYNAMICS:
+                  return placeAbove ? Sid::dynamicsPosAbove : Sid::dynamicsPosBelow;
+            case Tid::LYRICS_ODD:
+            case Tid::LYRICS_EVEN:
+                  return placeAbove ? Sid::lyricsPosAbove : Sid::lyricsPosBelow;
+            case Tid::REHEARSAL_MARK:
+                  return placeAbove ? Sid::rehearsalMarkPosAbove : Sid::rehearsalMarkPosBelow;
+            case Tid::STAFF:
+                  return placeAbove ? Sid::staffTextPosAbove : Sid::staffTextPosBelow;
+            case Tid::STICKING:
+                  return placeAbove ? Sid::stickingPosAbove : Sid::stickingPosBelow;
+            case Tid::SYSTEM:
+                  return placeAbove ? Sid::systemTextPosAbove : Sid::systemTextPosBelow;
+            case Tid::TEMPO:
+                  return placeAbove ? Sid::tempoPosAbove : Sid::tempoPosBelow;
+            default:
+                  break;
+            }
+      return Sid::NOSTYLE;
+      }
+
 //---------------------------------------------------------
 //   getPropertyStyle
 //---------------------------------------------------------
 
 Sid TextBase::getPropertyStyle(Pid id) const
       {
+      if (id == Pid::OFFSET) {
+            Tid defaultTid = Tid(propertyDefault(Pid::SUB_STYLE).toInt());
+            if (tid() == defaultTid) {
```

Comment:
... and this calculation is exactly what needs to use virtual functions mechanism here, but `propertyDefault()` seems to do this job well (and it is indeed a virtual function properly overridden in `TextBase` subclasses).